File: Spark/scala/WordCountRDDGCS.scala

import org.apache.spark._
import org.apache.spark.rdd._

object WordCountRDDGCS {
    def main(args : Array[String]) : Unit => {
        val sparkConf = new SparkConf()
            .setMaster("local[*]")
            .setAppName("WordCount_RDD_GCS_Scala")
            //choose one from the belowing
            //export GOOGLE_APPLICATION_CREDENTIALS=/home/to/your/credentials.json
            .set("GOOGLE_APPLICATION_CREDENTIALS", "/home/to/your/credentials.json")
        val sc = new SparkContext(sparkConf)
        
        val bucketName = args[0]
        val inputFileName = args[1]
        val outputFileName = args[2]
        val inputFilePath = "gs://" + bucketName + inputFileName
        val rdd = sc.textFile(inputFilePath)
        val word_count_rdd = rdd.flatMap(_.split(' '))
            .map(_.replaceAll("[^a-zA-Z]", "").toLowerCase)
            .filter(_ != "")
            .map((_, 1))
            .reduceByKey(_ + _)
            .sortBy(_._2, ascending=false)
        val top10 = word_count_rdd.take(10)

        top10.foreach(println)

        val outputFilePath = "gs://" + bucketName + outputFileName
        word_count_rdd.toDF().write.parquet(outputFilePath)
    }
}

File: Spark/python/word_count_rdd_gcs.py

from pyspark.sql import SparkSession

import re

spark = SparkSession.builder \
    .master('local[*]') \
    .appName('WordCount_RDD_GCS_Python') \
    .getOrCreate()
sc = spark.sparkContext

spark._jsc.hadoopConfiguration().set('google.cloud.auth.service.account.json.keyfile', '/home/to/your/credentials.json')

bucket_name = 'bucket_name'
input_file_name = '/textfile'
output_file_name = '/result'

input_file_path = 'gs://' + bucket_name + input_file_name
rdd = sc.textFile(input_file_path)
word_count_rdd = rdd.flatMap(lambda item: item.split(' ')) \
    .map(lambda item: re.sub(r'[^a-zA-Z]', '', item).lower()) \
    .filter(lambda item: item != '') \
    .map(lambda item: (item, 1)) \
    .reduceByKey(lambda x, y: x + y) \
    .sortBy(lambda x: x[1], ascending=False)
top10 = word_count_rdd.take(10)
print(top10)

output_file_path = 'gs://' + bucket_name + output_file_name
word_count_rdd.toDF().write.parquet(output_file_path)